src/sage/rings/power_series_pari.pyx

```diff
@@ -434,7 +434,7 @@ cdef class PowerSeries_pari(PowerSeries):
             if a.valuation() <= 0:
                 raise ValueError("can only substitute elements of positive valuation")
         elif isinstance(Q, PolynomialRing_generic):
-            Q = Q.completion(Q.gen())
+            Q = Q.completion(Q.variable_name())
```

Comment:
I wonder if this can be seen as a breaking change (previously it returns a power series, now it returns a `CompletionPolynomialRing`). The alternative is to special case such as `if p == self.gen():`